psi4/src/export_mints.cc

```diff
@@ -174,15 +174,16 @@ std::shared_ptr<BasisSet> construct_basisset_from_pydict(const std::shared_ptr<M
     auto basisset = std::make_shared<BasisSet>(key, mol, basis_atom_shell, basis_atom_ecpshell);
 
     // Modify the nuclear charges, to account for the ECP.
-    if (totalncore) {
+    if (key == "BASIS") {
```

Comment:
This makes sure Z is not modified for fitting basis sets? Should it not be applied to `"GUESS"` too?

Reply:
Well spotted. GUESS is a little different in that cast-up is handled with the 1st SCF / 2nd SCF sequence in scf_helper in proc.py, rather than being brought along as a fitting basis companion to the orbital. The orbital/basis, jkfit, rifit are the `key`s here and in https://github.com/psi4/psi4/blob/master/psi4/driver/qcdb/libmintsbasisset.py#L605. Whereas in the cast-up, the GUESS basis arrives here as a BASIS in its own right, possibly with JKFIT of its own.

I'm pretty sure of this, but if it's not clear or you still suspect I'm wrong, please say so.

Reply:
It's clear now that you explained it. Thanks!